common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.segment;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import com.google.common.collect.Lists;
+
+import org.apache.uniffle.common.BufferSegment;
+import org.apache.uniffle.common.ShuffleDataSegment;
+import org.apache.uniffle.common.ShuffleIndexResult;
+import org.apache.uniffle.common.exception.RssException;
+
+public class LocalOrderSegmentSplitter implements SegmentSplitter {
```

Comment:
Our blockId have the taskAttemptId information. We usually use the blockId to filter the data. You can see
 https://github.com/apache/incubator-uniffle/blob/47effb25044780013ff51cc516261464435b5829/client/src/main/java/org/apache/uniffle/client/impl/ShuffleReadClientImpl.java#L101

Reply:
So this splitter may be useless.

Reply:
OK. this will avoid using the startMapId and endMapId

Reply:
Let me think twice.

Reply:
What's the difference between LocalOrderSegmentSplitter?

Reply:
I have removed the startMapId and endMapId.

> What's the difference between LocalOrderSegmentSplitter and FixedSizeSegementSplitter?

LocalOrderSegmentSplitter is to get the segments according to the index file local order.

FixedSizeSegementSplitter is to keep the consistent with the original segment split logic

Reply:
In the future, maybe we could introduce the global order strategy which will be implemented in current abstraction easier.